### PR TITLE
Added Touch up,Touch move and Touch down handlers

### PR DESCRIPTION
**Pull request by anandsadu**
Added Touch up,Touch move and Touch down handlers.

### CODE REVIEW

File: app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        String path = "$.params.";
+        try {
+            JSONObject json = getPayload(request);
+            if (json.has("id")) {
+                String id = json.getString("id");
+                element = KnownElements.getElementFromCache(id);
+
+                if (element != null)
+                    isElement = true;
+                if (isElement) {
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: Is it possible to get here and not have an element? With the logic as it stands, if we get to here and there is no element, we will not populate `clickX` and `clickY` at all. Not sure what the consequences of that would be, though it seems like the touch event would not work.

Comment (anandsadu) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: Yeah..Here  I have added the exception for the missed case. If we have element we will get the bounds else we can get the x and y  coordinates from the json.
```
          if (json.has("id")) {
                String id = json.getString("id");
                element = KnownElements.getElementFromCache(id);
                if (element != null) { 
                       final Rect bounds = element.getBounds();
                       clickX = bounds.centerX();
                       clickY = bounds.centerY();
                 } else {
                    throw new UiObjectNotFoundException("Touch Event Element doesn't exist");
                    }
                }               
            } else { // no element so extract x and y from params
                clickX = JsonPath.compile(path + "x").read(json.toString());
                clickY = JsonPath.compile(path + "y").read(json.toString());
            }
```

Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: Where is this code from?

Comment (anandsadu) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: Referred [android bootstrap ](https://github.com/appium/appium-android-bootstrap/blob/master/bootstrap/src/io/appium/android/bootstrap/handler/TouchEvent.java#L57). I'll test the code with the changes mentioned in above comment and commit.

File: app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            if (json.has("id")) {
+                String id = json.getString("id");
+                element = KnownElements.getElementFromCache(id);
+
+                if (element != null) {
+                    final Rect bounds = element.getBounds();
+                    clickX = bounds.centerX();
+                    clickY = bounds.centerY();
+                } else {
+                    throw new UiObjectNotFoundException("TouchEvent element does not exist.");
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: Do you want to log the element id that was attempted?

File: app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    protected abstract boolean executeTouchEvent() throws UiObjectNotFoundException;
+
+    protected void printEventDebugLine(final String methodName,
+                                       final Integer... duration) {
+        String extra = "";
+        if (duration.length > 0) {
+            extra = ", duration: " + duration[0];
+        }
+        Logger.debug("Performing " + methodName + " using element? " + isElement
```
Comment (imurchie) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: `isElement` is no longer being set.

Comment (anandsadu) [inline app/src/main/java/io/appium/uiautomator2/handler/TouchEvent.java]: yeah...will remove that

**Pull request merged**